### PR TITLE
Adding script to convert NVIDIA nsys profiles to pbtxt

### PR DESCRIPTION
**Pull request by abhinavgoel95**
Adding a script that converts NVIDIA nsys profiles to a .pbtxt that is compatible with XLA's profile-guided latency estimator.

### CODE REVIEW

**Comment by abhinavgoel95** [comment]
@hawkinsp @nouiz @Tixxx please have a look.

File: jax/tools/pgo_nsys_converter.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+import csv
```
Comment (nouiz) [inline jax/tools/pgo_nsys_converter.py:1]: Can you add a comment here to tell what this does and how to use it?

Comment (hawkinsp) [reply jax/tools/pgo_nsys_converter.py:1]: I'd perhaps make it print meaningful help text, in addition to or partially in place of this!

File: jax/tools/pgo_nsys_converter.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+import sys
+import argparse
+import psutil
+import os
+import shutil
+import subprocess
+
+nsys_path = shutil.which("nsys")
+
+parser = argparse.ArgumentParser(description='Tool to sweep for optimal collective combiner threshold')
```
Comment (Tixxx) [inline jax/tools/pgo_nsys_converter.py:12]: We can change the description to just nsys->pbtxt conversion, this doesnt do sweeping anymore.

**Comment by nouiz** [comment]
This PR is ready for another review.

File: jax/tools/pgo_nsys_converter.py [hunk 3]
```diff
@@ -0,0 +1,47 @@
+import csv
```
Comment (hawkinsp) [inline jax/tools/pgo_nsys_converter.py:1]: I think you'll have to add the JAX copyright header (the apache license block) or one of our presubmits will get sad. Copy it from another file, update the year.

**Review by hawkinsp — CHANGES_REQUESTED** [review]

**Comment by abhinavgoel95** [comment]
@hawkinsp Added license information. Ready for review.

**Review by hawkinsp — APPROVED** [review]

**Comment by jakevdp** [comment]
This is failing a lint test (fixed in #20126) and also failing a doctest, because the script is part of the jax source tree, but doesn't check for `__main__` so the full script executes on import. Maybe we should wrap the main logic in `if __name__ == "__main__":`? What do you think?

**Comment by jakevdp** [comment]
I added the `__main__` guard in #20127

**Pull request merged**